packages/@dcl/inspector/src/lib/data-layer/host/rpc-methods.ts

```diff
@@ -1,14 +1,44 @@
 import { Entity, EntityMappingMode, IEngine, Composite, OnChangeFunction, CompositeDefinition } from '@dcl/ecs'
 
+import { saveEvent } from '../../../hooks/editor/useSave'
 import { DataLayerRpcServer, FileSystemInterface } from '../types'
 import { getFilesInDirectory } from './fs-utils'
 import { dumpEngineToComposite, dumpEngineToCrdtCommands } from './utils/engine-to-composite'
-import { createFsCompositeProvider } from './utils/fs-composite-provider'
+import { CompositeManager, createFsCompositeProvider } from './utils/fs-composite-provider'
 import { stream } from './stream'
 import { FileOperation, initUndoRedo } from './undo-redo'
 import { minimalComposite } from '../client/feeded-local-fs'
 import upsertAsset from './upsert-asset'
 
+function setupEngineDump(
+  fs: FileSystemInterface,
+  engine: IEngine,
+  compositeProvider: CompositeManager,
+  compositePath: string
+) {
+  return function dumpEngine(): CompositeDefinition {
+    // TODO: hardcoded for the moment
+    const composite = dumpEngineToComposite(engine, 'json')
+    // TODO: the ID should be the selected composite id name
+    // composite.id = 'main'
+
+    const mainCrdt = dumpEngineToCrdtCommands(engine)
+    fs.writeFile('main.crdt', Buffer.from(mainCrdt)).catch((err) => console.error(`Failed saving main.crdt: `, err))
+
+    compositeProvider
```

Comment:
await the promise for the rpc method
```suggestion
try {
  await compositeProvider.save({}))
} catch (e) {
// log
}
```

Reply:
I'm just moving it to another fn. This is already like this today because we are using it in a system, and systems cannot be thenables...